app/jobs/receive_submission_deliveries_job.rb

```
@@ -0,0 +1,74 @@
+class ReceiveSubmissionDeliveriesJob < ApplicationJob
+  require "aws-sdk-sqs"
+  require "aws-sdk-sts"
+
+  queue_as :background
+
+  REGION = "eu-west-2".freeze
+  SQS_QUEUE_NAME = "submission_email_ses_successful_deliveries_queue".freeze
+  MAX_NUMBER_OF_MESSAGES = 10
+  POLLING_PERIOD = 20
+
+  def perform
+    CurrentJobLoggingAttributes.job_id = job_id
+
+    sts_client = Aws::STS::Client.new(region: REGION)
+    queue_url = "https://sqs.#{REGION}.amazonaws.com/#{sts_client.get_caller_identity.account}/#{SQS_QUEUE_NAME}"
+    sqs_client = Aws::SQS::Client.new(region: REGION)
+
+    loop do
+      messages = receive_messages(sqs_client, queue_url)
+
+      break if messages.empty?
+
+      process_messages(sqs_client, queue_url, messages)
+    end
+  end
+
+private
+
+  def receive_messages(sqs_client, queue_url)
+    response = sqs_client.receive_message(
+      queue_url: queue_url,
+      max_number_of_messages: MAX_NUMBER_OF_MESSAGES,
+      wait_time_seconds: POLLING_PERIOD, # Any value > 0 enables long polling
+    )
+
+    Rails.logger.info("Received #{response.messages.length} messages - #{self.class.name}")
+
+    response.messages
+  end
+
+  def process_messages(sqs_client, queue_url, messages)
+    messages.each do |message|
+      sqs_message_id = message.message_id
+      CurrentJobLoggingAttributes.sqs_message_id = sqs_message_id
+
+      receipt_handle = message.receipt_handle
+      sns_message = JSON.parse(message.body)
+      ses_message = JSON.parse(sns_message["Message"])
+      ses_message_id = ses_message["mail"]["messageId"]
+      CurrentJobLoggingAttributes.mail_message_id = ses_message_id
+
+      ses_event_type = ses_message["eventType"]
+
+      submission = Submission.find_by!(mail_message_id: ses_message_id)
+
+      process_delivery(submission) if ses_event_type == "Delivery"
```

If the `eventType` isn't "Delivery" should we be deleting the message or should we instead throw an error? This shouldn't happen under regular circumstances but if we're checking the type at all maybe we shouldn't ignore it if it isn't what we expect.

---

Good thought, I suppose the same is true of the bounces and complaints job. A check that the `eventType` is what is expected and raising an error if not sounds best?